activestorage/lib/active_storage/analyzer/image_analyzer/image_magick.rb

```diff
@@ -32,5 +32,12 @@ def read_image
       def rotated_image?(image)
         %w[ RightTop LeftBottom TopRight BottomLeft ].include?(image["%[orientation]"])
       end
+
+      def opaque?(image)
+        return true unless image.data["channelDepth"].key?("alpha")
+
+        value = /7.\d/.match?(image.data["version"]) ? 255 : 0
+        image.data["channelStatistics"]["alpha"]["mean"] == value
```

Comment:
This is a lot of trial and error, and I have a problem here. For ImageMagick before 7 the value was `0`. For ImageMagick 7 and later the value is `255`. This PR works in both situations, but I don't know how to properly test for that in the CI.